src/toast/ToastProvider.tsx

```diff
@@ -0,0 +1,34 @@
+import React, {createContext, useReducer} from "react";
+
+import ToastStack from "./stack/ToastStack";
+import {initialToastState} from "./util/toastConstants";
+import toastReducer from "./util/toastReducer";
+import {ToastAction, ToastContextState} from "./util/toastTypes";
+
+const ToastContext = createContext<[ToastContextState, React.Dispatch<ToastAction>]>([
+  initialToastState,
+  () => undefined
+]);
+
+interface ToastContextProviderProps {
+  children: React.ReactNode;
```

Comment:
Currently we don't have a way to provide `customRootId` to ToastStack component. I think we can add that as a prop to the `ToastContextProvider` and users can provide the customRootId from there.